File: LovePlay-UIKit-Masonry/LovePlay/Class/Zone/Model/ZoneListModel.m
//
//  ZoneListModel.m
//  LovePlay
//
//  Created by Yuns on 2017/2/2.
//  Copyright © 2017年 Yuns. All rights reserved.
//

#import "ZoneListModel.h"

@implementation ZoneListModel

+ (NSDictionary *)modelContainerPropertyGenericClass
{
    return @{@"discuzList" : [ZoneDiscussDetail class]};
}

@end

@implementation ZoneDiscussDetail

+ (NSDictionary *)modelContainerPropertyGenericClass
{
    return @{@"detailList" : [ZoneDiscussItem class]};
}

@end

@implementation ZoneDiscussItem


@end

@implementation ZoneDiscussType

+ (NSDictionary *)modelCustomPropertyMapper
{
    return @{@"typeId" : @"id"};
}

@end

File: LovePlay-Swift3/LovePlay/Class/Zone/Model/ZoneListModel.swift
//
//  ZoneListModel.swift
//  LovePlay
//
//  Created by Yuns on 2017/6/4.
//  Copyright © 2017年 yuns. All rights reserved.
//

import UIKit
import HandyJSON

struct ZoneDiscussItem : HandyJSON {
    var bannerUrl : String?
    var discuzModelTypeId : String?
    var fid : String?
    var iconUrl : String?
    var modelDesc : String?
    var modelName : String?
    var todayPosts : String?
    var weight : String?
}

struct ZoneDiscussType : HandyJSON {
    var category : String?
    var typeId : String?
    var typeName : String?
    var weight : String?
    
    mutating func mapping(mapper: HelpingMapper) {
        mapper <<<
            typeId <-- "id"
    }
}

class ZoneListModel: HandyJSON {
    
    var detailList : [ZoneDiscussItem]?
    var type : ZoneDiscussType?
    
    required init() {
        
    }
}
